### ISSUE TITLE
Loss exploded???

### ISSUE THREAD

**Issue by WhiteFu** [rishikksh20/vae_tacotron2#2]
I get the error "loss explode" in the training stage! 
I'm not modifying the original hyperparameters, and I want to know how to solve the problem.

**Comment by adimukewar** [comment]
I am facing the same issue. Please let me know if you have resolved it.

**Comment by rishikksh20** [comment]
actually, it is a common occurrence when dealing with a variational autoencoder. Two way to resolve it 
1) again start training from 3 or 4 back saved checkpoints (not from recent one).
2) In the file train.py [https://github.com/rishikksh20/vae_tacotron2/blob/0ab0fb964be82111a0f57f6cec92d7e682fe20b1/tacotron/train.py#L133] change the value of the loss.
 @adimukewar @WhiteFu

**Comment by WhiteFu** [comment]
Thanks for your reply, I will take it immediately!

**Comment by WhiteFu** [comment]
> I am facing the same issue. Please let me know if you have resolved it.

sorry, I didn't reply to you in time. I have been trying some other work recently, so I haven't solved this problem

**Comment by rishikksh20** [comment]
@WhiteFu if you are using this code then use large (more than 50 hours) expressive dataset like a blizzard for getting a decent result.

**Comment by MisakaMikoto96** [comment]
> > I am facing the same issue. Please let me know if you have resolved it.
> 
> sorry, I didn't reply to you in time. I have been trying some other work recently, so I haven't solved this problem

hi, I have the same problem that I supposed to modified some hparams but it still not work.Please let me know if you have solved this. thx😄

**Comment by WhiteFu** [comment]
The loss is not stable, so you can modify the upper limit of the parameter In the file train.py on line 133,

**Comment by MisakaMikoto96** [comment]
> The loss is not stable, so you can modify the upper limit of the parameter In the file train.py on line 133,

hi, but it seems my loss = nan (every time at the same step when training)

**Comment by rishikksh20** [comment]
@MisakaMikoto96 aware of `Nan` loss, it means your variational autoencoder (VAE) unable to learn the latent representation. This is the common problem when you dealing with Variational autoencoder but the sad part is, there is no simple solution for that. 
One solution you can try to go [line](https://github.com/rishikksh20/vae_tacotron2/blob/0ab0fb964be82111a0f57f6cec92d7e682fe20b1/tacotron/utils/util.py#L24) and manipulate `w1` and `w2`. 
But before that make sure you have adequate quantity and expressiveness rich voice data and also sometimes after getting error, restart training from 2 steps back saved checkpoint is worked fine for me, if you getting the error again at the same checkpoint then restart from 3 steps back saved checkpoint and so on. If again and again, you get `Nan` at the same step count then try the above solution.
You can also read variational autoencoder paper for more understanding and otherwise feel free to ask here.
